ext/dba/dba.c

```diff
@@ -569,16 +568,15 @@ static void php_dba_open(INTERNAL_FUNCTION_PARAMETERS, bool persistent)
 		RETURN_THROWS();
 	}
 
-	char *resource_key;
-	size_t resource_key_len = spprintf(&resource_key, 0,
+	zend_string *resource_key = zend_strpprintf(0,
 		"dba_%d_%s_%s_%s", persistent, ZSTR_VAL(path), ZSTR_VAL(mode), handler_str ? ZSTR_VAL(handler_str) : ""
 	);
 
 	if (persistent) {
 		zend_resource *le;
 
 		/* try to find if we already have this link in our persistent list */
-		if ((le = zend_hash_str_find_ptr(&EG(persistent_list), resource_key, resource_key_len)) != NULL) {
+		if ((le = zend_hash_find_ptr(&EG(persistent_list), resource_key)) != NULL) {
 			if (le->type != le_pdb) {
 				// TODO This should never happen
 				efree(resource_key);
```

Comment:
This line must use zend_string_release_ex

Reply:
Ah indeed.